File: ReSharper.FSharp/src/Services.FSharp/src/Injected/RegExpPsiProvider.cs
using JetBrains.ProjectModel;
using JetBrains.ReSharper.Plugins.FSharp.Psi;
using JetBrains.ReSharper.Psi.Impl.Shared.InjectedPsi;
using JetBrains.ReSharper.Psi.RegExp.ClrRegex;

namespace JetBrains.ReSharper.Plugins.FSharp.Services.Cs.Injected
{
  [SolutionComponent]
  public class RegExpPsiProvider : LiteralsInjectionPsiProvider<FSharpLanguage, ClrRegexLanguage>
  {
    public RegExpPsiProvider(FSharpInjectionProvider injectorProvider)
      : base(injectorProvider, ClrRegexLanguage.Instance)
    {
    }

    public override bool ProvidedLanguageCanHaveNestedInjects => false;
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Injected/RegExpPsiProvider.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Injected

open JetBrains.ProjectModel
open JetBrains.ReSharper.Plugins.FSharp.Psi
open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Injected
open JetBrains.ReSharper.Psi.Impl.Shared.InjectedPsi
open JetBrains.ReSharper.Psi.RegExp.ClrRegex

[<SolutionComponent>]
type RegExprPsiProvider(injectorProvider: FSharpRegexInjectionProvider) =
    inherit LiteralsInjectionPsiProvider<FSharpLanguage, ClrRegexLanguage>(injectorProvider, ClrRegexLanguage.Instance)

    override _.ProvidedLanguageCanHaveNestedInjects = false
